### PR TITLE
Add tests where circle is not located in the first quadrant and tests with small voxels

### PR DESCRIPTION
**Pull request by nukenukenukelol**
Add test cases for positive x&y and negative x&y for grid located at second, third, fourth quadrant and origin. 
Also, add a test case with small voxeles to see if small voxel would break our algorithm. Currently, the test case has 20 radial sec and 20 angular sec.

### CODE REVIEW

**Comment by nukenukenukelol** [comment]
Currently, I believe our polar algo is failing some previous test cases and the small voxel test

**Comment by ak-2485** [comment]
Can you be more specific? When I run the tests I get that 20 pass, 2 fail. The two that fail we discussed as being ambiguous.

**Comment by ak-2485** [comment]
Can you edit the PR so that it doesn't include the changes to the Readme?

**Comment by nukenukenukelol** [comment]
> ambiguous

Hmmmm it could be my polar algo version is not most updated, I uploaded the newest version online before run the tests. Maybe something didn't work out. I will check again

File: matlab/polar-traversal/polarCoordinateTraversalTest.m [hunk 1]
```diff
@@ -524,3 +524,227 @@ function testSimulhits(testCase)
...
+        expected_rVoxels     = [1,2,3,3,2,1];
+        expected_thetaVoxels = [2,2,2,1,1,1];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
+% The center of the circle is on the forth quadrant with negative x and y
+% direction
+function testForthQuadrantNegativeXNegativeY(testCase)
```
Comment (cgyurgyik) [inline matlab/polar-traversal/polarCoordinateTraversalTest.m]: Fourth

File: matlab/polar-traversal/polarCoordinateTraversalTest.m [hunk 2]
```diff
@@ -524,3 +524,227 @@ function testSimulhits(testCase)
...
+        expected_rVoxels     = [1,2,2,1];
+        expected_thetaVoxels = [0,0,1,1];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
+% The center of the circle is on the forth quadrant with positive x and y
+% direction
+function testForthQuadrantPositiveXPositiveY(testCase)
```
Comment (cgyurgyik) [inline matlab/polar-traversal/polarCoordinateTraversalTest.m]: Fourth

**Comment by nukenukenukelol** [comment]
I also include test over the cord length test in the updated version of polarCoordinateTraversal.m. I apologize for the confusion over the pull request, I should start the request on a new branch.

File: matlab/polar-traversal/polarCoordinateTraversal.m [hunk 3]
```diff
@@ -188,34 +188,53 @@
...
-        current_voxel_ID_r = current_voxel_ID_r + tStepR;
-    else 
-       return;
-    end 
-        angular_voxels = [angular_voxels, current_voxel_ID_theta];
-        radial_voxels = [radial_voxels, current_voxel_ID_r];
+% Initialize a variable to accumulate of time it takes for all traversals. This 
+% will be used to compare with the cord length (t2 - t1) we calculated earlier 
+if verbose
+  acc = 0.0;
```
Comment (nukenukenukelol) [inline matlab/polar-traversal/polarCoordinateTraversal.m]: The incrementer

**Comment by nukenukenukelol** [comment]
So, we are still failing Fourth Quadrant Negative X Negative Y. It seems that the algo fails to find the last hit? I change t_end to 100 but the problem is still there. I'm not 100% sure if my test case is wrong or something else is happening
Screen Shot 2020-04-01 at 3 43 53 PM

**Comment by cgyurgyik** [comment]
Please revert the spacing changes you made in polarCoordinateTraversal.m to make the diff more readable

File: matlab/polar-traversal/polarCoordinateTraversalTest.m [hunk 4]
```diff
@@ -524,3 +524,227 @@ function testSimulhits(testCase)
...
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [1,1,2,1];
+        expected_thetaVoxels = [2,1,1,1];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
+% The center of the circle is on the origin of the grid with negative x and
```
Comment (cgyurgyik) [inline matlab/polar-traversal/polarCoordinateTraversalTest.m:703]: This test fails. Here is what I get:
```
rVoxels =

     1     1     2     1


thetaVoxels =

     2     1     1     1
```
Furthermore, the ray direction is positive in the x- and y-directions. I'm assuming this is a typographical error somewhere?

**Comment by ak-2485** [comment]
What about t_end = min(t_grid, t_end) + tol; ?

File: matlab/polar-traversal/polarCoordinateTraversalTest.m [hunk 5]
```diff
@@ -524,3 +524,227 @@ function testSimulhits(testCase)
...
+    max_bound = [30.0, 30.0];
+    ray_origin = [15.0, 9.0];
+    ray_direction = [-1.3, -0.5];
+    circle_center = [0.0, 0.0];
+    circle_max_radius = 10.0;
+    num_radial_sections = 20;
+    num_angular_sections = 20;
+    t_begin = 0.0;
+    t_end = 30.0;
+    verbose = true
```
Comment (cgyurgyik) [inline matlab/polar-traversal/polarCoordinateTraversalTest.m:741]: Remove

**Comment by cgyurgyik** [comment]
> What about t_end = min(t_grid, t_end) + tol; ?
In polarCoordinateTraversal, ```tol = 10^-15```. This doesn't work.
10^-14 does works however.

**Comment by cgyurgyik** [comment]
Issue found with small voxel count:
Screen Shot 2020-04-02 at 11 00 31 AM

```
thetaVoxels =  1     2     2     2     2     2     2     2     2     2     3     
                         3     3     3     3     4     4     5     6     7     7     8     
                         8     8     9     9     9     9     9     9    9     9    10    
                         10    10    10
```
The algorithm produces thetaVoxel as being 1 during initialization phase (using both old version with atan2 and new version with no trigonometric function), yet this image clearly depicts it being 2 (unless I'm making some small error here).

File: matlab/polar-traversal/polarCoordinateTraversalTest.m (hunk 4 above)
Comment (nukenukenukelol) [reply matlab/polar-traversal/polarCoordinateTraversalTest.m:703]: I just updated the fixed version of NegativeXNegativeY test

**Comment by cgyurgyik** [comment]
Can you submit your cord changes in a separate PR please? The tests look good and I want to merge them, but I don't know about the cord test.
